### DISCUSSION TITLE
How to export data in CSV or Json?

### DISCUSSION THREAD

**Discussion by ViktorVarjas** [mobilise-d/mobgap#178] [Q&A] [answered]
Hi Everyone!

When having a code snippet like below, running a pipeline, how to export the results in CSV or Json? 

Thank you in advance!
Viktor

```
data = load_data(inFile)

pipeline = MobilisedPipelineUniversal()
dataset = CustomDataset(data)

pipeline = pipeline.safe_run(dataset)

print(pipeline.aggregated_parameters_)
print(pipeline.per_wb_parameters_)
print(pipeline.per_stride_parameters_)

#ToDo: add saving results

print(f'Processed file: {file_name}')

**Comment by AKuederle** [comment, marked as answer]
Hi @ViktorVarjas, All the returned values are Pandas Dataframes. So you can use all the Pandas native methods to export results. Have a look at [this page](https://pandas.pydata.org/docs/reference/io.html) for ways to export Pandas dataframes.

Btw. If your input data structure is complex enough that you want to use a Custom Dataset, it might make sense to also include the data loading logic directly. See [this example](https://mobgap.readthedocs.io/en/latest/auto_examples/data/_05_custom_datasets.html#sphx-glr-auto-examples-data-05-custom-datasets-py) for options on how to handle custom data

**Reply by ViktorVarjas** [reply]
Thank your very much for the quick response! Saving results work fine.

I'm trying to process data from an Axivity sensor in CSV using this Custom data and load function to convert the data as required:
I hope it makes sense what I'm doing here :D

```
class CustomDataset:
    sampling_rate_hz: float = 100
    measurement_condition = "free_living"
    recording_metadata = {"measurement_condition": "free_living"}

    def __init__(self, data):
        self.data = data

    @property
    def participant_metadata(self):
        return {
            "sensor_height_m": 1.0,
            "height_m": 1.60,
            "cohort": "HA",
        }
        
    @property
    def data_ss(self):
        return self.data[['time', 'acc_x', 'acc_y', 'acc_z', 'gyr_x', 'gyr_y', 'gyr_z']]  

    def get_data(self):
        return self.data
    
#this converts the data from g to m/s^2 (acc * 9.80665)
def load_data(path: Path) -> pd.DataFrame:
    data = pd.read_csv(path)
    data[["acc_x", "acc_y", "acc_z"]] = (data[["acc_x", "acc_y", "acc_z"]] * GRAV_MS2)
    return data

**Reply by AKuederle** [reply]
Totally fine to do it the way you did. You could have done that without creating a custom class using the `[GaitDatasetFromData](https://mobgap.readthedocs.io/en/latest/modules/generated/data/mobgap.data.GaitDatasetFromData.html#mobgap.data.GaitDatasetFromData)` class. It allows you to simply pass the and metadata in the constructor. The idea was, that this could be used exactly for the usecase you have here. In case, this does not work for you, let me know! Would be intersted to make your usecase easier.

Custom datasets "shine", when you have a whole bunch of recordings that you want to process or if one file that you want to load contains multiple recordings. Then you can build a custom datasets to load and format the data on the fly while still allowing for concepts like a grid search or cross validation across your pool of recordings
